clang/test/CodeGen/bounds-checking-debuginfo.c

```diff
@@ -89,7 +89,7 @@ double f1(int b, int i) {
 // CHECK-TRAP: [[DBG21]] = !DILocation(line: 65, column: 3, scope: [[DBG4]])
 // CHECK-TRAP: [[DBG22]] = !DILocation(line: 66, column: 12, scope: [[DBG4]])
 // CHECK-TRAP: [[DBG23]] = !DILocation(line: 0, scope: [[META24:![0-9]+]], inlinedAt: [[DBG26]])
-// CHECK-TRAP: [[META24]] = distinct !DISubprogram(name: "__ubsan_check_array_bounds", scope: [[META5]], file: [[META5]], type: [[META25:![0-9]+]], flags: DIFlagArtificial, spFlags: DISPFlagDefinition, unit: [[META0]])
+// CHECK-TRAP: [[META24]] = distinct !DISubprogram(name: "__ubsan_check_array-bounds", scope: [[META5]], file: [[META5]], type: [[META25:![0-9]+]], flags: DIFlagArtificial, spFlags: DISPFlagDefinition, unit: [[META0]])
```

Comment:
can we break some some DI processing tools if we use "-" in symbols?

Reply:
yeah, that too. i think we should sanitize the function names

Reply:
The part of patch wich is not NFC for existing tools, like bounds, better to pre-commit in a separate PR

Reply:
to be more clear: i don't think we should put `-` in function names. something invariably will break somewhere because it doesn't expect it.

Reply:
Ack-nowledged, will-fix

Reply:
StringSanitizer added in https://github.com/llvm/llvm-project/pull/139809/commits/dfc3919acb799b7ebb0b99f7003b3cd6097d8ea7